### PR TITLE
Fix test TestServer

### PR DESCRIPTION
**Pull request by tc-hib**
* typo in comments
* loop variable captured in function literal

See https://golang.org/doc/faq#closures_and_goroutines

### CODE REVIEW

File: sse_test.go [hunk 1]
```diff
@@ -63,6 +63,7 @@ func TestServer(t *testing.T) {
 			id := fmt.Sprintf("C-%d", n+1)
 			fmt.Printf("Client %s registed to channel %s\n", id, name)
 
+			name := name
```
Comment (alexandrevicenzi) [inline sse_test.go:66]: What are you trying to fix here? This was done in a purpose, the exact one that you pointed is the doc, it's not wrong.

**Comment by tc-hib** [comment]
See this example: [playground](https://play.golang.org/p/nkyz1jBaQDl)
You can run `go vet` too, and see the warning.
You might want to have look at [this issue](https://github.com/golang/go/issues/20733#issue-237009049) too.

In a nutshell, if you capture a loop variable in an inner goroutine, you can't predict what value it will have on actual execution. Most of the time you'll only get a few samples repeated many times.

**Comment by alexandrevicenzi** [comment]
Yeah, but you do not fix the issue by removing the code, you should pass name to the goroutine to fix it properly, the same way as done by id.

**Comment by tc-hib** [comment]
I'm sorry I don't understand.
What code did I remove?
I only added a shadow variable.

File: sse_test.go (hunk 1 above)
Comment (tc-hib) [reply sse_test.go:66]: image

**Comment by alexandrevicenzi** [comment]
I'm probably missing coffee, I thought you removed the code, but you added. Now things do make sense.

Yet, if you pass the variable to the goroutine, it looks cleaner than redefining it. Perhaps you can change it.

**Comment by tc-hib** [comment]
image

**Comment by alexandrevicenzi** [comment]
I'm not looking for easier, I'm looking for a cleaner solution. What someone would put an eye on and get right at it, redefining it is something very Go specific, that from Python, Java or other language perspective makes zero sense.

**Comment by tc-hib** [comment]
Do as you wish, it's your project.
But it seems to be the idiomatic Go way of solving this issue, easier to read and maintain (no parameter order).
This how Goland solves it automatically too.
It's already a closure anyway, so why make it more complicated?

**Comment by alexandrevicenzi** [comment]
id is already a parameter, leaving name outside is not concise. Two ways of doing the same thing in the same scope.

**Pull request closed without merging**